### PR TITLE
adding a header with variables

### PR DESCRIPTION
**Pull request by thasaarah**
work in progress with function that does it automatically

### CODE REVIEW

File: models/header.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+def add_header(file_name: str, parameter: dict):
+    """
+    create a header introducing the values associated with each variable used in the file
+
+    Parameters:
+    ---------
+    file: str
+        name of the file without the header
+    parameters: dict
```
Comment (Ipuch) [inline models/header.py:9]: same here and every in the script

File: models/header.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        dictionary pairing each parameter with its default value
+
+    """
+    # name the new file
+    new_file_name = file_name.removesuffix(".bioMod") + "_with_variables.bioMod"
```
Comment (Ipuch) [inline models/header.py:14]: "replace new_file_name " by "new_biomod_file_name"

Comment (Ipuch) [inline models/header.py:14]: "replace new_file_name " by "biomod_file_name"

File: models/header.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    biomod_file = open(new_file_name, 'w')
+
+    # copy the first line of the original file onto the new file
+    biomod_file = open(new_file_name, 'w')
```
Comment (Ipuch) [inline models/header.py:19]: is it necessary to do it twice? if not delete one of the lines.

File: models/header.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+thorax_values = {
+    'thoraxRT1': 1.5136279122166798,
+    'thoraxRT2': 0.02508004601838823,
+    'thoraxRT3': -0.12026902121482706,
+    'thoraxRT4': -0.5902166829985935,
+    'thoraxRT5': 0.6147618471392384,
+    'thoraxRT6': 0.35577136570126267
+}
+
+add_header(file, thorax_values)
```
Comment (Ipuch) [inline models/header.py:49]: Finally, I think this is better to put the new file name as an argument of the function. Can you modify this?

**Review by Ipuch — COMMENTED** [review]
Nice work! :+1: 

There are a few details to adjust. Answer "Done" to the comments when they are treated.

Also, modify the script's name with the primary function "add_header" as utils and put the example in another script named examples.py

Final work, modify the file README.md in the folder models to add a description of the new .biomod added.

File: models/header.py (hunk 2 above)
Comment (thasaarah) [reply models/header.py:14]: Done

File: models/header.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+def add_header(file_name: str, parameter: dict):
```
[the remark this answers is not in the document]
Comment (thasaarah) [reply models/header.py:1]: Done

File: models/header.py (hunk 1 above)
Comment (thasaarah) [reply models/header.py:9]: Done

File: models/header.py (hunk 3 above)
Comment (thasaarah) [reply models/header.py:19]: Done

File: models/header.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (thasaarah) [reply models/header.py:49]: Done

**Comment by Ipuch** [comment]
Let's merge this :+1:

**Pull request merged**